MapStruct 1.5.0.Beta1 ZonedDateTime field mapping regression

Hi team, thanks for the hard work - 1.5.0.Beta1 looks awesome! Meanwhile I've found one regression as following:

This was working with 1.4.1.Final:

```java
package test;

import java.time.ZonedDateTime;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper
interface TestMapper {
  @Mapping(target = "dateTime", source = ".", dateFormat = "YYYY-MM-DD")
  Test map(String s);

  class Test {

    public ZonedDateTime getDateTime() {
      return dateTime;
    }

    public void setDateTime(ZonedDateTime dateTime) {
      this.dateTime = dateTime;
    }

    ZonedDateTime dateTime;
  }
}
```

but started to fail with 1.5.0.Beta1:
```
TestMapperImpl.java:23: error: cannot find symbol
            test.setDateTime( ZonedDateTime.parse( s, dateTimeFormatter_YYYY_MM_DD_11271985664 ) );
                                                      ^
  symbol:   variable dateTimeFormatter_YYYY_MM_DD_11271985664
  location: class TestMapperImpl
```

Compiler:
```
ᐅ javac -version
javac 11.0.11
ᐅ java -version
openjdk version "11.0.11" 2021-04-20
OpenJDK Runtime Environment AdoptOpenJDK-11.0.11+9 (build 11.0.11+9)
OpenJDK 64-Bit Server VM AdoptOpenJDK-11.0.11+9 (build 11.0.11+9, mixed mode)
```

--- a/processor/src/main/java/org/mapstruct/ap/internal/conversion/ReverseConversion.java
+++ b/processor/src/main/java/org/mapstruct/ap/internal/conversion/ReverseConversion.java
@@ -9,6 +9,7 @@ import java.util.Collections;
 import java.util.List;
 import org.mapstruct.ap.internal.model.common.Assignment;
 import org.mapstruct.ap.internal.model.common.ConversionContext;
+import org.mapstruct.ap.internal.model.common.FieldReference;
 import org.mapstruct.ap.internal.model.HelperMethod;
 
 /**
@@ -44,4 +45,8 @@ public class ReverseConversion implements ConversionProvider {
         return Collections.emptyList();
     }
 
+    @Override
+    public List<FieldReference> getRequiredHelperFields(ConversionContext conversionContext) {
+        return conversionProvider.getRequiredHelperFields( conversionContext );
+    }
 }

--- a/processor/src/test/java/org/mapstruct/ap/test/bugs/_2530/Issue2530Mapper.java
+++ b/processor/src/test/java/org/mapstruct/ap/test/bugs/_2530/Issue2530Mapper.java
@@ -1,0 +1,37 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.bugs._2530;
+
+import java.time.LocalDate;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.factory.Mappers;
+
+/**
+ * @author Ben Zegveld
+ */
+@Mapper
+public interface Issue2530Mapper {
+
+    Issue2530Mapper INSTANCE = Mappers.getMapper( Issue2530Mapper.class );
+
+    @Mapping(target = "date", source = ".", dateFormat = "yyyy-MM-dd")
+    Test map(String s);
+
+    class Test {
+
+        LocalDate date;
+
+        public LocalDate getDate() {
+            return date;
+        }
+
+        public void setDate(LocalDate dateTime) {
+            this.date = dateTime;
+        }
+    }
+}

--- a/processor/src/test/java/org/mapstruct/ap/test/bugs/_2530/Issue2530Test.java
+++ b/processor/src/test/java/org/mapstruct/ap/test/bugs/_2530/Issue2530Test.java
@@ -1,0 +1,34 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.bugs._2530;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.Month;
+
+import org.mapstruct.ap.testutil.IssueKey;
+import org.mapstruct.ap.testutil.ProcessorTest;
+import org.mapstruct.ap.testutil.WithClasses;
+
+/**
+ * @author Ben Zegveld
+ */
+@IssueKey("2530")
+@WithClasses({
+    Issue2530Mapper.class
+})
+public class Issue2530Test {
+
+    @ProcessorTest
+    public void shouldConvert() {
+        Issue2530Mapper.Test target = Issue2530Mapper.INSTANCE.map( "2021-07-31" );
+
+        assertThat( target ).isNotNull();
+        assertThat( target.getDate().getYear() ).isEqualTo( 2021 );
+        assertThat( target.getDate().getMonth() ).isEqualTo( Month.JULY );
+        assertThat( target.getDate().getDayOfMonth() ).isEqualTo( 31 );
+    }
+}

